dask/bag/core.py

```diff
@@ -244,20 +267,53 @@ def __init__(self, dsk, name, npartitions):
         self.npartitions = npartitions
         self.str = StringAccessor(self)
 
-    def map(self, func):
+    def map(self, func, **kwargs):
         """ Map a function across all elements in collection
 
         >>> import dask.bag as db
         >>> b = db.from_sequence(range(5))
         >>> list(b.map(lambda x: x * 10))  # doctest: +SKIP
         [0, 10, 20, 30, 40]
+
+        Any **kwargs are passed through to func along with the (maybe unpacked)
+        item argument(s):
+
+        >>> import dask.bag as db
+        >>> b = db.from_sequence(range(100), npartitions=10)
+        >>> def normalize(x, total=0):
+        ...     return x / total
+        >>> total = b.sum().compute()
+        >>> total
+        4950
+        >>> normalized = b.map(normalize, total=total)
+        >>> normalized.sum().compute()
+        1.0
+
+        If a dask.Bag.Item is used its computed value is passed to the map
+        function.  This means that whatever computations are implied by any
+        Item values block any of the map work.
+
+        For example the above ended up doing its two passes over the data in
+        two separate dask computations.  If the data were expensive to access
+        (rather than being an in-memory range) we would instead want to combine
+        both passes over the data into one shared computation:
+
+        >>> normalized = b.map(normalize, total=b.sum())
+        >>> normalized.sum().compute()
+        1.0
         """
-        name = 'map-{0}-{1}'.format(funcname(func), tokenize(self, func))
+        name = 'map-{0}-{1}'.format(funcname(func),
+                                    tokenize(self, func, kwargs))
         if takes_multiple_arguments(func):
             func = partial(apply, func)
-        dsk = dict(((name, i), (reify, (map, func, (self.name, i))))
+        dsk = self.dask.copy()
+        if kwargs:
+            kw_dsk, kw_pairs = unpack_kwargs(kwargs)
+            dsk = merge(dsk, kw_dsk)
```

Comment:
Same here. You can just mutate `dsk` instead of creating a new one. Not super important, as time difference will be negligible.